src/main.cpp

```diff
@@ -3854,9 +3859,34 @@ bool ReceivedBlockTransactions(
                 } else {
                     pindex->nChainSaplingValue = std::nullopt;
                 }
+
+                // Given that all parents are BLOCK_VALID_TRANSACTIONS, we now should be
+                // able to calculate this block's net transparent value, assuming the
+                // block is correct.
+                CAmount transparentValue = 0;
+                for (auto tx : block.vtx) {
+                    for (auto out : tx.vout) {
+                        transparentValue += out.nValue;
+                    }
+                    if (view.HaveInputs(tx)) {
+                        transparentValue -= view.GetTransparentValueIn(tx);
+                    } else {
+                        // Some of the inputs are inaccessible; this might be a block from
+                        // a fork that spends coins also spent in the current chain. Fall
+                        // back to directly looking up the transactions.
+                        // TODO
+                    }
+                }
+                pindex->nTransparentValue = transparentValue;
```

Comment:
This section actually needs to be before the `if (pindex->pprev) {` conditional instead of inside it, so that the genesis block `nChainTransparentValue` is correctly set.

Reply:
Moved at https://github.com/zcash/zcash/pull/4470/commits/3d7f7ae52e757ce74ea60d766a4fd8dde29b6341